core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -19,14 +21,14 @@ import scala.jdk.CollectionConverters._
 import scala.math.BigDecimal.RoundingMode
 
 object SpendReportingService {
-  def constructor(dataSource: SlickDataSource, bigQueryDAO: GoogleBigQueryDAO, samDAO: SamDAO, spendReportingServiceConfig: SpendReportingServiceConfig)
+  def constructor(dataSource: SlickDataSource, bigQueryService: cats.effect.Resource[IO, GoogleBigQueryService[IO]], samDAO: SamDAO, spendReportingServiceConfig: SpendReportingServiceConfig)
```

Comment:
Why does this need to be wrapped with `cats.effect.Resource` (I don't know much about cats yet)?

Reply:
Also, I find reading cats-effect stuff to be really difficult and unwelcoming, so definitely let me know if you'd like to talk through any of it since I've found that to be more helpful at times.